src/uthash.h

```diff
@@ -146,8 +158,12 @@ do {
 do {                                                                             \
   (tbl)->bloom_nbits = HASH_BLOOM;                                               \
   (tbl)->bloom_bv = (uint8_t*)uthash_malloc(HASH_BLOOM_BYTELEN);                 \
-  if (!(tbl)->bloom_bv) {                                                        \
-    uthash_fatal("out of memory");                                               \
+  if (!((tbl)->bloom_bv))  {                                                     \
+    if (HASH_NOMEM_OK) {                                                         \
+      _mem_ok = 0;                                                               \
```

Comment:
Basically yes. Think about all these macros as if they were C functions; what you're doing here is analogous to using a global variable. (Actually it's worse, because it's a global variable that may or may not exist depending on who called me.)  In C we'd fix that by passing in the address of the variable to modify as an `int*` function parameter; in macro-land we fix it by passing in the *name* of the variable to modify as a macro parameter.